app/forms/steps/outgoings/board_and_lodgings_form.rb

```diff
@@ -0,0 +1,53 @@
+module Steps
+  module Outgoings
+    class BoardAndLodgingsForm < Steps::BaseFormObject
+      attribute :board_amount, :pence
+      attribute :food_amount, :pence
+      attribute :frequency, :value_object, source: PaymentFrequencyType
+      attribute :payee_name, :string
+      attribute :payee_relationship_to_client, :string
+
+      validates :board_amount, presence: true
+      validates :food_amount, presence: true
+      validates :frequency, inclusion: { in: PaymentFrequencyType.values }
+      validates :payee_name, presence: true
+      validates :payee_relationship_to_client, presence: true
+
+      def outgoings_payment
+        @outgoings_payment ||= crime_application.outgoings_payments.find_by(
+          payment_type: OutgoingsPaymentType::BOARD_AND_LODGING.to_s
+        )
+      end
+
+      # extract the values from OutgoingsPayment.details hash if they already exist
+      # and use to set the form attributes on load
+      def self.build(crime_application)
+        @outgoings_payment ||= crime_application.outgoings_payments.find_by(
+          payment_type: OutgoingsPaymentType::BOARD_AND_LODGING.to_s
+        )
+
+        form = new
+        form.frequency = @outgoings_payment.frequency
+        @outgoings_payment.details&.each do |attr_name, value|
+          form.send("#{attr_name}=", value)
+        end
+        form
+      end
+
+      private
+
+      def amount
+        amount = board_amount.to_f - food_amount.to_f
+        amount.to_s
+      end
+
+      def persist!
+        outgoings_payment.update!(
+          amount: amount,
+          frequency: frequency,
+          details: attributes.except('frequency')
```

Comment:
Would prefer the attributes to be saved to be explicitly listed (allowlist) rather than a denylist.

Reply:
Okay good point, will update this!